security-ldap/src/main/java/io/micronaut/configuration/security/ldap/LdapAuthenticationProvider.java

```diff
@@ -175,8 +187,11 @@ public LdapAuthenticationProvider(LdapConfiguration configuration,
                 }
             } finally {
                 contextBuilder.close(managerContext);
+                emitter.onComplete();
```

Comment:
This cannot be moved here because its not possible for both onError and onComplete to be called

Reply:
I was not aware that after `onError` it was not necessary to called `onComplete`. I have reverted the change.